pkg/rbac/parser.go

```diff
@@ -121,45 +121,64 @@ func (Generator) RegisterMarkers(into *markers.Registry) error {
 	return into.Register(RuleDefinition)
 }
 
-func (g Generator) Generate(ctx *genall.GenerationContext) error {
-	rules := make(map[ruleKey]*Rule)
+// GenerateClusterRole generates a rbacv1.ClusterRole object
+func GenerateClusterRole(ctx *genall.GenerationContext, roleName string) (*rbacv1.ClusterRole, error) {
+	// Using a slice here to preserve the order of Rules
+	var ruleSlice []Rule
+	// ruleKeyToIndexMap maps a ruleKey to its index in ruleSlice
+	ruleKeyToIndexMap := make(map[ruleKey]int)
```

Comment:
You don't need a map now.

Reply:
I kept the map, because the map is useful to track whether a Rule with a certain key has existed or not.